src/main/java/com/zaxxer/nuprocess/internal/BasePosixProcess.java

```diff
@@ -82,6 +92,26 @@
 
    private Pointer posix_spawn_file_actions;
 
+   // Launches threads under which changes to cwd do not affect the cwd of the process.
+   public static class LinuxCwdChangeableThreadFactory implements ThreadFactory {
+      private final AtomicInteger threadCount = new AtomicInteger(0);
+
+      @Override
+      public Thread newThread(final Runnable r) {
+         Runnable unshareCwdThenSpawn = new Runnable() {
+            @Override
+            public void run() {
+              int rc = LinuxLibC.unshare(LinuxLibC.CLONE_FS);
```

Comment:
Makes since on the ``unshare()``.

With respect to the exception, I think it is going to be swallowed by the executor, so there needs to be a way to propagate it out so that it is eventually thrown from the ``start()`` method.